src/snowflake/snowpark/mock/_plan.py

```diff
@@ -1444,7 +1444,13 @@ def calculate_expression(
                 raise_error=NotImplementedError,
             )
         try:
-            return input_data[exp.name]
+            if exp.name in input_data:
```

Comment:
Does this mean there is a mixture of quoted and unquoted names here? How do we distinguish between a column that is explicitly quoted? For example:
```
df = session.create_dataframe(
    [[1,2],[3,4]], schema=['col1', '"col2"']
)

# df['col1'1] should work
# df['"col1"'] should not work
# df['"col2"'] should work
# df['col2'] should not work
```

Reply:
good catch, after doing some test, snowflake always add double quotes to the col name when loading json.

I removed the logic in the _plan.py, and add the logic of handling column name (add double quotes/escape double quotes) to the read json function.